machinery/comparator.go

```diff
@@ -201,6 +202,20 @@ func (d *Comparator) Compare(
 		return res, fmt.Errorf("API accessor: %w", err)
 	}
 
+	// If there is a Status subresource defined, add .status to stripset to ignore it.
+	stripSet := stripSet
```

Comment:
`stripSet` is a pointer to a `fieldpath.Set` - why is it captured here?

Reply:
Because I am overriding the pointer in the if block below, if I would not capture the variable, I'd change the global `stripSet`.
I've made this clearer by renaming the variable `localStripSet` in the next commit.

Reply:
That makes sense. Thanks for the explanation and making it clearer in the code! :)